docs: Add plugin access methods to iOS SDK docs (#707)
amplitude/amplitude-docs: content/collections/ios_sdk/en/ios-swift-sdk.md

{{/partial:tab}}
{{/partial:tabs}}

### Accessing plugins

> You can access plugins that have been added to the Amplitude instance by name or by type.

Reviewer: ```suggestion
Access plugins that are added to the Amplitude instance by name or by type.
```

if (myPlugin != nil) {
    // Use the plugin
}
```
{{/partial:tab}}
{{/partial:tabs}}

#### Access plugins by type

To access all plugins of a specific type, use the `plugins(type:)` method:

{{partial:tabs tabs="Swift, Obj-C"}}
{{partial:tab name="Swift"}}
```swift
// Define a custom plugin type
class MyEnrichmentPlugin: Plugin {
    let name = "my-enrichment-plugin"
    let type = PluginType.enrichment
    
    func setup(amplitude: Amplitude) {
        // Setup code
    }
    
    func execute(event: BaseEvent?) -> BaseEvent? {
        // Execute code
        return event
    }
}

// Add multiple plugins of the same type
amplitude.add(plugin: MyEnrichmentPlugin())
amplitude.add(plugin: AnotherEnrichmentPlugin())

// Later, access all plugins of a specific type
let enrichmentPlugins = amplitude.plugins(type: MyEnrichmentPlugin.self)
for plugin in enrichmentPlugins {
    // Use each plugin of the specified type
}
```
{{/partial:tab}}
{{partial:tab name="Obj-C"}}
```objc
// Objective-C doesn't support the plugins(type:) method directly
// You would need to use the plugin(name:) method to access specific plugins
```
{{/partial:tab}}
{{/partial:tabs}}

> This method returns an array of all plugins that match the specified type, allowing you to interact with multiple plugins of the same type at once.

Reviewer: ```suggestion
This method returns an array of all plugins that match the specified type, allowing you to interact with more than one plugin of the same type.
```
